docs/writing-policies/go/04-validation.md

```diff
@@ -8,331 +8,420 @@ title: ""
 	// NOTE 4
-	data.ForEach(func(key, value gjson.Result) bool {
-		label := key.String()
-
-		// NOTE 5
-		if settings.DeniedLabels.Contains(label) {
-			err = fmt.Errorf("Label %s is on the deny list", label)
-			// stop iterating over labels
-			return false
-		}
-
-		// NOTE 6
-		regExp, found := settings.ConstrainedLabels[label]
-		if found {
-			// This is a constrained label
-			if !regExp.Match([]byte(value.String())) {
-				err = fmt.Errorf("The value of %s doesn't pass user-defined constraint", label)
-				// stop iterating over labels
-				return false
-			}
-		}
+	// Try to create a Pod instance using the RAW JSON we got from the
+	// ValidationRequest.
+	pod := &corev1.Pod{}
+	if err := easyjson.Unmarshal([]byte(podJSON), pod); err != nil {
+		return kubewarden.RejectRequest(
+			kubewarden.Message(
+				fmt.Sprintf("Cannot decode Pod object: %s", err.Error())),
+			kubewarden.Code(400))
+	}
 
-		return true
+	logger.DebugWithFields("validating pod object", func(e onelog.Entry) {
+		e.String("name", pod.Metadata.Name)
+		e.String("namespace", pod.Metadata.Namespace)
 	})
 
-	// NOTE 7
-	if err != nil {
-		return kubewarden.RejectRequest(
-			kubewarden.Message(err.Error()),
-			kubewarden.NoCode)
+	// highlight-next-line
+	// NOTE 5
+	for label, value := range pod.Metadata.Labels {
+		if err := validateLabel(label, value, &settings); err != nil {
+			return kubewarden.RejectRequest(
+				kubewarden.Message(err.Error()),
+				kubewarden.NoCode)
+		}
 	}
 
 	return kubewarden.AcceptRequest()
 }
 ```
 
-The code has some `NOTE` sections inside of it. Let's get through them:
-
-1. The function ensures the JSON payload is properly formatted. This
-  is done using a function provided by the `gjson` library
-2. The `Settings` instance is created using one of the constructor
-  methods we defined inside of `settings.go`
-3. We use a `gjson` selector to get the `label` map provided by the object
-  embedded into the request
-4. We use a `gjson` helper to iterate over the results of the query. If the query
-  has no results the loop will never take place.
-5. We look for the `label` of the object inside of the list of denied labels
-  provided by the user via the policy settings. If the `label` is a denied
-  one, we set the value of the `err` variable and exit from the loop (that
-  happens by returning `false` instead of `true`).
-6. We look for the `label` of the object inside of the list of constrained labels
-  provided by the user via the policy settings. When we have a match we use
-  the regular expression provided by the user to validate the value of the label.
-  If the validation fails, we set the value of the `err` variable and exit
-  from the loop (that happens by returning `false` instead of `true`).
-7. If the `err` variable is not `nil`, we use the helper provided by Kubewarden's
-  SDK to reject the request. Otherwise we accept it.
+The code has some `NOTE` sections inside of it, let's get through them:
```

Comment:
Should we be coloring these note sections as code? Or some other particular format?
```suggestion
The code has some `NOTE` sections inside of it, let's get through them:
```